Entities/SpikeJumpThroughController.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.Xna.Framework;
+using Monocle;
+using Celeste.Mod.SpringCollab2020;
+using Celeste.Mod.Entities;
+using System.Linq;
+
+// Based on RainbowSpinnerColorController
+// from SpringCollab2020 / MaxHelpingHand
+
+namespace Celeste.Mod.SpringCollab2020.Entities {
+    [CustomEntity("SpringCollab2020/SpikeJumpThroughController")]
+    class SpikeJumpThroughController : Entity {
+        private static bool SpikeHooked;
+        private static SpikeJumpThroughController CurrentController;
+        private static SpikeJumpThroughController NextController;
+        private static float TransitionProgress = -1f;
```

Comment:
Storing the current controller, next controller and transition progress was useful for the rainbow spinner color controller to fade between 2 values during screen transitions. In your case it might be simpler to do something similar to Parallax Fade Out Controller (which is also a controller that applies a hook if it is in the room): https://github.com/max4805/MaxHelpingHand/blob/master/Entities/ParallaxFadeOutController.cs

(Keep the hook on Load level though, it's useful to make the controller persist when you save & quit)